From bcf3418306b8c9f9cf10b128052ed10e726657f7 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 26 Oct 2018 19:16:21 +0800
Subject: [PATCH] rk: arm64: support build [z]boot.img with ramdisk

The ramdisk name should be ramdisk.img.
The ramdisk.img should be placed on $(objtree).
Support symbolic link.

Change-Id: Iee78c437bf8291362adf9598578d69de6d6abab6
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm64/Makefile | 28 ++++++++++++++++------------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 6edd48b23f1b..74a89b4cf64c 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -172,22 +172,26 @@ endif
 
 LOGO := $(notdir $(wildcard $(srctree)/logo.bmp))
 LOGO_KERNEL := $(notdir $(wildcard $(srctree)/logo_kernel.bmp))
+RAMDISK_IMG := $(notdir $(wildcard $(objtree)/ramdisk.img))
+make_boot_img := \
+	$(srctree)/scripts/mkbootimg \
+		--kernel $(objtree)/arch/arm64/boot/Image \
+		$(if $(RAMDISK_IMG),--ramdisk $(objtree)/ramdisk.img) \
+		--second resource.img \
+		-o boot.img && \
+	echo '  Image:  boot.img (with Image $(if $(RAMDISK_IMG),'ramdisk.img ')resource.img) is ready'; \
+	$(srctree)/scripts/mkbootimg \
+		--kernel $(objtree)/arch/arm64/boot/Image.lz4 \
+		$(if $(RAMDISK_IMG),--ramdisk $(objtree)/ramdisk.img) \
+		--second resource.img \
+		-o zboot.img && \
+	echo '  Image:  zboot.img (with Image.lz4 $(if $(RAMDISK_IMG),'ramdisk.img ')resource.img) is ready'
+
 %.img: rockchip/%.dtb kernel.img $(LOGO) $(LOGO_KERNEL)
 	$(Q) if [ -f $(srctree)/$(LOGO) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO) $(objtree)/; fi
 	$(Q) if [ -f $(srctree)/$(LOGO_KERNEL) -a "$(srctree)" != "$(objtree)" ]; then cp -a $(srctree)/$(LOGO_KERNEL) $(objtree)/; fi
 	$(Q)scripts/resource_tool $(objtree)/arch/arm64/boot/dts/rockchip/$*.dtb $(LOGO) $(LOGO_KERNEL)
 	@echo '  Image:  resource.img (with $*.dtb $(LOGO) $(LOGO_KERNEL)) is ready'
-	$(Q) if [ -x $(srctree)/scripts/mkbootimg ]; then \
-		$(srctree)/scripts/mkbootimg \
-			--kernel $(objtree)/arch/arm64/boot/Image \
-			--second resource.img \
-			-o boot.img && \
-		echo '  Image:  boot.img (with Image resource.img) is ready'; \
-		$(srctree)/scripts/mkbootimg \
-			--kernel $(objtree)/arch/arm64/boot/Image.lz4 \
-			--second resource.img \
-			-o zboot.img && \
-		echo '  Image:  zboot.img (with Image.lz4 resource.img) is ready'; \
-	fi
+	$(Q) if [ -x $(srctree)/scripts/mkbootimg ]; then $(make_boot_img); fi
 
 CLEAN_FILES += boot.img kernel.img resource.img zboot.img
-- 
2.35.3

